Only support stats.Cumulative window for Prometheus

odeke-em opened the pull request:
If a View doesn't have a stats.Cumulative
window at export time, skip collecting it.

Fixes #214

rakyll commented on exporter/stats/prometheus/prometheus.go:
```diff
@@ -170,14 +170,14 @@ func (c *collector) Describe(ch chan<- *prometheus.Desc) {
-	}
-	c.mu.Unlock()
-
 	for _, vd := range c.viewData {
```
Can we reject them at Export instead?

odeke-em replied:
I was following suit with https://github.com/census-instrumentation/opencensus-go/blob/9b9df23b5a299ba044805cdd2c28764702cfddab/exporter/stats/stackdriver/stackdriver.go#L168-L172 and the suggestion from the issue originally, but for sure, it even makes it easier to reject them in the Export method.

odeke-em commented:
@rakyll I've also added non Cumulative View rejection in Export by proxy of method addView. I extracted the code into a helper and we perform that check at two different call sites:
* Collect
* Export
PTAL

rakyll commented on exporter/stats/prometheus/prometheus_test.go:
```diff
@@ -0,0 +1,82 @@
+		},
+		2: {
+			vds: &stats.ViewData{
+				View: newView(stats.CountAggregation{}, fauxWindow{}),
```
Just use Interval window instead.

odeke-em replied:
Nice, thanks. Updating that in a second.

odeke-em commented:
Thank you for the review @rakyll, I addressed your feedback, tests passing and I got your blessing with a LGTM, so I'll merge.
